cmd/client-s3.go

```diff
@@ -223,6 +219,30 @@ func newFactory() func(config *Config) (Client, *probe.Error) {
 				}
 			}
 
+			var creds *credentials.Credentials
+
+			// if an STS endpoint is set, we will use that for credentials
+			stsEndpoint := env.Get("STS_ENDPOINT", "")
+			if stsEndpoint != "" {
+				stsEndpointURL, err := url.Parse(stsEndpoint)
+				if err != nil {
+					return nil, probe.NewError(fmt.Errorf("Error parsing sts endpoint: %v", err))
+				}
+				creds = credentials.New(&credentials.IAM{
+					Client: &http.Client{
+						Transport: transport,
+					},
+					Endpoint: stsEndpointURL.String(),
+				})
```

Comment:
may I suggest an improvement?

```
//      creds := credentials.NewChainCredentials(
//          []credentials.Provider{
//              &credentials.IAM{},
//              credentials.NewStaticV4(config.AccessKey, config.SecretKey, config.SessionToken),
//              credentials.NewStaticV2(config.AccessKey, config.SecretKey, ""),
//          })
```

use Chain credentials

Reply:
How do these work? it tries them in that order?

Reply:
Yeah read the code you will understand :-)

Reply:
implemented